sdk/storage/Azure.Storage.Common/src/StorageCrc64Calculator.cs

```diff
@@ -0,0 +1,1052 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Buffers.Binary;
+
+namespace Azure.Storage
+{
+    /// <summary>
+    /// Port of https://github.com/Azure/azure-storage-net/blob/master/Lib/Common/Core/Util/Crc64.cs
+    /// and the code used by Azure Storage to compute a CRC-64 with custom polynomial.
+    /// </summary>
+    internal static class StorageCrc64Calculator
```

Comment:
do we have extensive tests for this class?

Reply:
I can search the v11 library for tests but this is a copy of a copy of the actual code in the service. I'm not entirely certain what would even be tested for something like this. We may need to reach out to FE if the v11 library just took their word for it.